azext_iot/central/providers/device_provider.py

```diff
@@ -114,7 +119,7 @@ def create_device(
         )
 
         if not device:
-            raise CLIError("No device found with id: '{}'.".format(device_id))
+            raise ResourceNotFoundError("No device found with id: '{}'.".format(device_id))
```

Comment:
Hrm, in this case we are in the context of create device, a falsey result from the service component indicates a failure. Two comments here:
- The exception text should say something like "Failure to create device with id ....".
- Since we don't have the response exception here a `AzureResponseError` could be appropriate to represent a generic service error response.